src/storm/mesos/logviewer/SocketUrlDetection.java

```diff
@@ -0,0 +1,66 @@
+package storm.mesos.logviewer;
+
+import java.net.InetAddress;
+import java.net.Socket;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.log4j.Logger;
+
+import com.google.common.base.Optional;
+
+public class SocketUrlDetection implements IUrlDetection {
+
+    public static final int DEFAULT_PORT = 8000;
+    
+    public SocketUrlDetection() {
+        setPort(Optional.of(DEFAULT_PORT));
+    }
+    
+    public SocketUrlDetection(int port) {
+        setPort(Optional.of(port));
+    }
+    
+    public SocketUrlDetection(Optional<Integer> port) {
+        setPort(port);
+    }
+    
+    @Override
+    public boolean isReachable() {
+
+        Socket socket       = null;
+        boolean reachable   = false;
+        try {
+            
+            if (LOG.isDebugEnabled())
+                LOG.debug("Checking host " + InetAddress.getLocalHost() + " and port " + getPort() );
+            
+            if (port.isPresent()) {
+                socket      = new Socket(InetAddress.getLocalHost(), getPort().get());
+                reachable   = true;
+            }
+        } catch (Exception e) {
+            // don't care.
+        }
+        finally {
+            IOUtils.closeQuietly(socket);
+        }
+        return reachable;
+    }
+    
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    @Override
+    public void setPort(Optional port) {
```

Comment:
The reason its and Optional and should stay that way is because the IUrlDetection interface requires an Optional. And as previously stated the one implementation of that interface uses an Optional. It sort of all is integrated and plays together. At this point I feel it should stay as is.